transcendence-app/src/users/users.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+
+// This should be a real class/interface representing a user entity
+export type UserEntity = {
```

Comment:
Why do you need to redefine this type if you have already a class defined with this same info (createUserDto)

Reply:
I don't really know the standard way of working with these classes and Dtos. We have the Dto class to create and validate our user, but maybe we shoud store it in a UserEntity class. Maybe it is a duplicating a little but the purpose of those classes are different. But I would have to go deeper into the documentation or examples to give a better answer.

Reply:
why is this a type and not a class? Why is it commented as class/interface?

Reply:
This service is incomplete and more like a mock service. For now, the user entities are stored in an array.

A future goal is to store the users in a database.
We could inject a repository to access the database and manage user entities.

Reply:
Agree, but I think you are using dtos instead of entites for storing userData :see_no_evil:

Reply:
I've removed the user entity since there is no database yet. We can add it back later if we need it.